File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long int count = 0;
        long long int mn = INT_MAX;
        long long int sum = 0;
        for(long long int i=0;i<matrix.size();i++){
            for(long long int j=0;j<matrix[i].size();j++){
                mn = fmin(mn,abs(matrix[i][j]));
                if(matrix[i][j] <= 0) count++;
                sum += abs(matrix[i][j]);
            }
        }
        return count % 2 == 0 ? sum : sum - ((mn)+(mn));
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
import numpy as np
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        ip = np.array(matrix)
        ans = ip[ip > 0]
        ans2 = ip[ip <= 0]
        ans3 =np.abs(ip)
        # ans3= ip[(lambda ip:abs(ip))]
        # print(ans3)
        # print(ans, ans2)
        if np.shape(ans2)[0] % 2 == 0:
            return int(ans.sum() + abs(ans2.sum()))
        else:
            # min =np.array(abs(ans2.max()),ans.min())
            # print(min)
            
            return int(ans.sum() + abs(ans2.sum()) - (2 *ans3.min() ))
